client/question/services/code_evaluators/PythonCodeRunnerService.js

```diff
@@ -89,9 +107,13 @@ tie.factory('PythonCodeRunnerService', [
         }
 
         // The run was successful.
-        return CodeEvalResultObjectFactory.create(
-          code, outputLines.join('\n'), observedOutputs,
+        var standardizedOutput = _standardizeOutputInClient(
+          outputLines, separator);
+        var codeEvalResult = CodeEvalResultObjectFactory.create(
+          code, standardizedOutput, observedOutputs,
           buggyOutputTestResults, performanceTestResults, null, null);
+        _displayPrintedOutput(codeEvalResult);
```

Comment:
This call seems to be in the wrong place. This is a code runner, it shouldn't be displaying anything. It should just run the code and return the evaluation result.